src/foam/nanos/script/BeanShellExecutor.java

```diff
@@ -0,0 +1,38 @@
+package foam.nanos.script;
+
+import java.io.IOException;
+import java.io.PrintStream;
+
+import bsh.EvalError;
+import bsh.Interpreter;
+import foam.core.X;
+import foam.nanos.boot.NSpec;
+import foam.nanos.logger.Logger;
+
+public class BeanShellExecutor {
+  private NSpec nSpec;
+  public static final Object[] OBJECT_HOLDER = new Object[1];
+  
+  public BeanShellExecutor(NSpec nSpec) {
+    this.nSpec = nSpec; 
+  }
+
+  public Object execute(X x, PrintStream ps, String serviceScript ) throws IOException  {
+    Interpreter shell = new Interpreter();
+    try {
+      shell.set("x", x);
+      Object service = shell.eval(serviceScript);
+      nSpec.saveService(x, service);
```

Comment:
Either remove this side effect that doesn't do anything anyway (saveService code is commented out), or it should also be added to the JShellExecutor. A better way to do this is to move the saveService call back to the NSpec.createService method instead of trying to do it here.

Reply:
Yes, leave this to the caller.